File: CustomKeyboard/app/src/main/java/com/cdc/Keyboard/basic/BaseActivity.java
package com.cdc.keyboard.basic;

import android.os.Bundle;
import com.cdc.keyboard.basic.api.BasePresenter;

public abstract class  BaseActivity<P extends BasePresenter> extends RootActivity{
    protected P mPresenter;
    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        initCreate(savedInstanceState);
        mPresenter = setPresenter();
        mPresenter.attachView(this);
        mPresenter.setContext(this);
        initView();
        initData();
    }
    protected abstract void initCreate(Bundle savedInstanceState);
    protected abstract void initView();
    protected abstract void initData();
    protected abstract P setPresenter();



    @Override
    protected void onDestroy() {
        super.onDestroy();
        mPresenter.detachView();
    }
}

File: CustomKeyboard/app/src/main/java/com/cdc/Keyboard/basic/BaseActivity.kt
package com.cdc.keyboard.basic

import android.content.Intent
import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import com.cdc.keyboard.R

open class BaseActivity : AppCompatActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_base)
    }

    protected  fun <T> goToPage(cla:Class<T>){
        val intent = Intent(this, cla)
        startActivity(intent)
    }
}
